packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -303,8 +316,7 @@ function querySelectorPatched(this: Element /*, selector: string*/): Element | n
             );
             return isUndefined(elm) ? null : elm;
         } else {
-            const elm = nodeList[0];
-            return isUndefined(elm) ? null : elm;
+            return nodeList.length === 0 ? null : nodeList[0];
```

Comment:
Am I missing something or does this mean the same thing as before?

Reply:
the same thing, @ravijayaramappa says this way reveals more intent.